frontend/test/playwright/utils/navigation.ts

```diff
@@ -95,203 +88,170 @@ export const searchTypeNames = {
   },
 }
 
-const isButtonPressed = async (
+/**
+ * On mobile screen, open the "tab" in the content settings modal.
+ * Should be called after the modal is open.
+ */
+export const openContentSettingsTab = async (
   page: Page,
-  buttonSelector: string
-): Promise<boolean> => {
-  const viewportSize = page.viewportSize()
-  if (!viewportSize) {
-    return false
-  }
-  const pageWidth = viewportSize.width
-  if (pageWidth > 640) {
-    return await getPressed(page, buttonSelector)
-  } else {
-    return await page.locator("button", { hasText: "Close" }).isVisible()
-  }
-}
+  tab: "searchTypes" | "filters" = "searchTypes",
+  dir: LanguageDirection = "ltr"
+) => {
+  const tabKey = tab === "searchTypes" ? "search-type.heading" : "filters.title"
 
-const openMenu = async (page: Page, button: "filter" | "contentSwitcher") => {
-  const selector = buttonSelectors[button]
-  if (!(await isButtonPressed(page, selector))) {
-    await page.click(selector)
-    expect(await isButtonPressed(page, selector)).toEqual(true)
-  }
+  await page.getByRole("tab", { name: t(tabKey, dir) }).click()
 }
 
-export const openFilters = async (
+/**
+ * On mobile screen, clicks on the "Close" button in the content settings modal.
+ */
+export const closeContentSettingsModal = async (
   page: Page,
   dir: LanguageDirection = "ltr"
 ) => {
-  if (isPageDesktop(page)) {
-    await openMenu(page, "filter")
-  } else {
-    await openContentSettingsTab(page, "filters", dir)
-  }
+  return page
+    .getByRole("button", { name: t("modal.close-content-settings", dir) })
+    .click()
 }
 
-export const openContentTypes = async (
+/**
+ * Opens or closes the search settings:
+ * - given modal tab on mobile screen
+ * - filters sidebar or the search types popover on desktop screen
+ */
+export const setContentSwitcherState = async (
   page: Page,
+  contentSwitcherKind: "filters" | "searchTypes",
+  state: "open" | "closed",
   dir: LanguageDirection = "ltr"
 ) => {
-  if (isPageDesktop(page)) {
-    await openMenu(page, "contentSwitcher")
-  } else {
-    await openContentSettingsTab(page, "contentTypes", dir)
+  const isDesktop = isPageDesktop(page)
+
+  const buttonLocator = page.locator(
+    !isDesktop
+      ? "#content-settings-button"
+      : contentSwitcherKind === "filters"
+      ? "#filter-button"
+      : "#search-type-button"
+  )
+
+  const isPressed = await getSelectorPressed(buttonLocator)
+  const shouldBePressed = state === "open"
+
+  if (isDesktop) {
+    if (isPressed === shouldBePressed) return null
+    return await buttonLocator.click()
+  }
+
+  if (shouldBePressed) {
+    if (!isPressed) {
+      await buttonLocator.click()
+    }
+    return openContentSettingsTab(page, contentSwitcherKind, dir)
+  } else if (isPressed) {
+    await closeContentSettingsModal(page, dir)
   }
 }
 
+export const filters = {
```

Comment:
Nice! I like this object-grouped model. In the future maybe the cookies could be handled similarly.